targets/XPlatCppSdk/source/code/include/playfab/PlayFabGameServerSDK.h

```diff
@@ -0,0 +1,71 @@
+    {
+        class PlayFabGameServerSDK
+        {
+        public:
+            /// <summary>Renamed from WaitForSessionAssignment. Called when the game server is ready to accept clients.  If Start() hasn't been called by this point, it will be called implicitly here.</summary>
+            /// <remarks>Required. This is a blocking call and will only return when this server is either allocated (a player is about to connect) or terminated.</remarks>
+            /// <returns>True if the server is allocated (will receive players shortly). False if the server is terminated. </returns>
+            static bool readyForPlayers();
+
+            /// <summary>Returns all configuration settings</summary>
+            /// <returns>unordered map of string key:value configuration setting values</returns>
+            static const std::unordered_map<std::string, std::string> &getConfigSettings();
+
+            /// <summary>Kicks off communication threads, heartbeats, etc.  Called implicitly by ReadyForPlayers if not called beforehand.</summary>
+            /// <param name="debugLogs">Enables outputting additional logs to the GSDK log file.</param>
+            static void start(bool debugLogs = false);
+
+            /// <summary>Tells the Xcloud service information on who is connected.</summary>
+            /// <param name="currentlyConnectedPlayers"></param>
+            static void updateConnectedPlayers(const std::vector<GameServerConnectedPlayer> &currentlyConnectedPlayers);
+
+            /// <summary>Gets called if the server is shutting us down</summary>
+            static void registerShutdownCallback(std::function<void()> callback);
+
+            /// <summary>Gets called when the agent needs to check on the game's health</summary>
+            static void registerHealthCallback(std::function<bool()> callback);
+
+            /// <summary>Gets called if the server is getting a scheduled maintenance, it will get the UTC time of the maintenance event as an argument.</summary>
+            static void registerMaintenanceCallback(std::function<void(const tm &)> callback);
+
+            /// <summary>outputs a message to the log</summary>
+            static unsigned int logMessage(const std::string &message);
+
+            /// <summary>Returns a path to the directory where logs will be mapped to the VM host</summary>
+            static const std::string &getLogsDirectory();
+
+            /// <summary>Returns a path to the directory shared by all game servers to cache data.</summary>
+            static const std::string &getSharedContentDirectory();
+
+            /// <summary>After allocation, returns a list of the initial players that have access to this game server, used by PlayFab's Matchmaking offering</summary>
+            static const std::vector<std::string> &getInitialPlayers();
+
+            // Keys for the map returned by getConfigSettings
+
+            static constexpr const char* HEARTBEAT_ENDPOINT_KEY = "gsmsBaseUrl";
```

Comment:
>gsmsBaseUrl [](start = 67, length = 11)

can we make these the same as C# (non-legacy values)?

Reply:
I knew I was forgetting something! yes, I'll change them

---
In reply to: [256915647](https://github.com/PlayFab/SDKGenerator/pull/402#discussion_r256915647) [](ancestors = 256915647)